### PR TITLE
[Android] fix accessibility of Picker

### LINKED ISSUE #3454: Picker accessibility text is wrong
### Description

Using voiceover/talkback on both Android and iOS, picker control is read as "entry/textfield, double tap to edit",
also adding AutmationProperies.SetName + AutomationProperties.SetHelpText does not prevent device from reading texts in this order:
1. AutomationProperties.SetName 
2. "textfield"
3. AutomationProperties.SetHelpText
4. "double tap to edit"

Additionally, focusing picker (either by method call or by tapping) causes voiceover/talkback to say "character mode, something something"

### Expected Behavior
Read picker as a Button. Picker has no meaning in accessibility (especially on Android). The TextField that's clicked on should read as a "Button" and not a text field because the action a user is going to take on the picker will be the same as a button.  When a user hears that something is an "edit text field" that tells the user the interaction is going to be via a keyboard.

Using SetName and SetHelpText also needs to be more resilient on Android which will probably involve injecting yourself into the delegates that are used to build the help text used.

https://stackoverflow.com/questions/25641081/android-acccessibility-how-do-i-change-the-text-read-out-loud-for-an-edittext-v

Also once something is selected the "Text Field" needs to retain whatever was set for SetName and SetHelpText. 

### Actual Behavior
picker is read as textfield/entry

### Basic Information

- Version with issue: 3.1.0.583944
- IDE: VS for Mac Comunity 2017 
- Platform Target Frameworks: 
  - iOS: 11.4
  - Android: 27
- Android Support Library Version: v7

### PR DESCRIPTION
**Pull request by paymicro**
### Description of Change ###

Changed accessibility of Picker. Now it reads like a `button`. If something is selected in it, this item is also read. This is part of PR https://github.com/xamarin/Xamarin.Forms/pull/3955

### Issues Resolved ### 

- fixes for Android #3454 

### API Changes ###
 
 None

### Platforms Affected ### 

- Android

### Behavioral/Visual Changes ###

Picker supports accessibility parameters without changing hint/placeholder of control.

### Before/After Screenshots ### 

Before
image

After
image

### Testing Procedure ###

- run UItest 3454
- check accessibility of Pickers

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by PureWeen** [comment]
@paymicro 
- rebase to fix conflicts I'm curious to tests these with material :-) 
- I have one question about what it reads back. After you select something from the picker it reads the text "Control Gallery" in the middle of reading back about the button.  That doesn't seem quite right. Otherwise it looks like it's working great!!

**Comment by paymicro** [comment]
@PureWeen done! Please re-review. 
- With material is working correctly.
- Reading "Control Gallery" after closing dialog not fixes. Apparently, we need another fix for closing the dialog.

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3454.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
+﻿#if APP
```
Comment (PureWeen) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3454.cs]: These aren't needed if you aren't using a XAML file

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3454.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+				new Picker(),
+				new Picker { Title = "Title1" },
+				new Picker { Title = "Title2" },
+				new Picker { Visual = VisualMarker.Material },
+				new Picker { Visual = VisualMarker.Material },
+				new Picker { Title = "Title1", Visual = VisualMarker.Material },
+				new Picker { Title = "Title2", Visual = VisualMarker.Material },
+			};
+
+			pickers[0].SetAutomationPropertiesName("First accessibility");
```
Comment (PureWeen) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3454.cs]: I noticed that with these items it doesn't read what's selected in the box. If you don't have these specified it'll read out what's selected first

- click box
- select item 6
- click off box
- click back on box (here it should read "item 6" and then the rest of the text)

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3454.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+			// material
+			pickers[4].SetAutomationPropertiesName("First accessibility");
+			pickers[4].SetAutomationPropertiesHelpText("This is the accessibility text");
+			pickers[7].SetAutomationPropertiesName("Last accessibility");
+			pickers[7].SetAutomationPropertiesHelpText("This is the accessibility text");
+
+			pickers.ForEach(p => p.ItemsSource = Enumerable.Range(1, 10).Select(i => $"item {i}").ToList());
+
+			Content = new StackLayout
```
Comment (PureWeen) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3454.cs]: Can you wrap this in a scrollview?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3454.cs (hunk 1 above)
Comment (paymicro) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3454.cs]: It is needed that the auto test is not crashed on the `SetAutomationPropertiesName`. I'll move them closer `AutomationProperties` to  to be more clear.

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3454.cs (hunk 2 above)
Comment (paymicro) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3454.cs]: Fixed

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3454.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 3454, "Picker accessibility text is wrong", PlatformAffected.Android)]
+	public class Issue3454 : TestContentPage
+	{
+		protected override void Init()
+		{
+			var pickers = new List<Picker>();
+			var grid = new Grid();
+			int row = 0, col = 0;
+			int index = 0;
```
Comment (samhouts) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3454.cs]:
```suggestion
```

**Pull request merged**